outlines/models/openai.py

```diff
@@ -468,3 +412,57 @@ def call(*args, **kwargs):
             raise e
 
     return call
+
+
+def openai(
+    model_name: str,
+    api_key: Optional[str] = None,
+    system_prompt: Optional[str] = None,
+    config: Optional[OpenAIConfig] = None,
+):
+    try:
+        import tiktoken
+        from openai import AsyncOpenAI
+    except ImportError:
+        raise ImportError(
+            "The `openai` and `tiktoken` libraries needs to be installed in order to use Outlines' OpenAI integration."
+        )
+
+    if config is not None:
+        config = replace(config, model=model_name)  # type: ignore
+    else:
+        config = OpenAIConfig(model=model_name)
+
+    client = AsyncOpenAI(api_key=api_key)
+    tokenizer = tiktoken.encoding_for_model(model_name)
+
+    return OpenAI(client, config, tokenizer)
+
+
+def azure_openai(
+    model_name: str,
+    azure_endpoint: Optional[str] = None,
+    api_version: Optional[str] = None,
+    api_key: Optional[str] = None,
+    system_prompt: Optional[str] = None,
+    config: Optional[OpenAIConfig] = None,
+):
+    try:
+        import tiktoken
+        from openai import AzureAsyncOpenAI
+    except ImportError:
+        raise ImportError(
+            "The `openai` and `tiktoken` libraries needs to be installed in order to use Outlines' Azure OpenAI integration."
+        )
+
+    if config is not None:
+        config = replace(config, model=model_name)  # type: ignore
```

Comment:
In the `azure.py` file, it used to be `self.config = replace(config, model=deployment_name)`, so I think the model should be the deployment_name